certora/specs/Reverts.spec

```diff
@@ -0,0 +1,283 @@
+
+    // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
+    require collateralQuoted > 0;
+
+    mathint borrowed = require_uint256(tAU(pBorrowShares, mTotalBorrowAssets, mTotalBorrowShares));
+    mathint ltv = require_uint256(wDU(require_uint256(borrowed), require_uint256(collateralQuoted)));
+
+    preLiquidate@withrevert(e, borrower, seizedAssets, 0, data);
+
+    assert !priceChanged && (ltv <= currentContract.PRE_LLTV) => lastReverted;
+}
+
+rule excessivePreliquidationReverts(env e,address borrower, uint256 seizedAssets, bytes data) {
+    uint256 mTotalBorrowAssets;
+    uint256 mTotalBorrowShares;
+    uint256 mLastUpdate;
+
+    uint256 pBorrowShares;
+    uint256 pCollateral;
+
+    (_, _, mTotalBorrowAssets,mTotalBorrowShares,mLastUpdate, _) = MORPHO.market(currentContract.ID);
+
+    // Ensure that no interest is accumulated.
+    require mLastUpdate == e.block.timestamp;
+
+    // Consider that the collateral price hasn't changed.
+    priceChanged = false;
+
+    uint256 collateralPrice = mockPrice();
+
+    (_, pBorrowShares, pCollateral) = MORPHO.position(currentContract.ID, borrower);
+
+    mathint collateralQuoted = require_uint256(summaryMulDivDown(pCollateral, collateralPrice, ORACLE_SCALE()));
+
+    // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
+    require collateralQuoted > 0;
+
+    requireInvariant preLltvLTlltv();
+    requireInvariant preLCFIncreasing();
+    requireInvariant preLIFIncreasing();
+
+    mathint borrowed = require_uint256(tAU(pBorrowShares, mTotalBorrowAssets, mTotalBorrowShares));
+    mathint ltv = require_uint256(wDU(require_uint256(borrowed), require_uint256(collateralQuoted)));
+
+
+    mathint preLIF = computeFactor(ltv,
+                                   currentContract.LLTV,
+                                   currentContract.PRE_LLTV,
+                                   currentContract.PRE_LIF_1,
+                                   currentContract.PRE_LIF_2) ;
+
+    // Safe require as implementation would revert.
+    require seizedAssets > 0;
+
+    mathint seizedAssetsQuoted = require_uint256(summaryMulDivUp(seizedAssets, collateralPrice, ORACLE_SCALE()));
+
+    mathint repaidShares = tSU(wDU(require_uint256(seizedAssetsQuoted), require_uint256(preLIF)),
+                               mTotalBorrowAssets,
+                               mTotalBorrowShares);
+
+    mathint closeFactor = computeFactor(ltv,
+                                        currentContract.LLTV,
+                                        currentContract.PRE_LLTV,
+                                        currentContract.PRE_LCF_1,
+                                        currentContract.PRE_LCF_2) ;
+
+    mathint repayableShares = wMD(pBorrowShares, require_uint256(closeFactor));
+
+    preLiquidate@withrevert(e, borrower, seizedAssets, 0, data);
+    assert (!priceChanged && repaidShares > repayableShares ) => lastReverted;
```

Comment:
we could require !priceChanged

It allows to put a comment about why it's fine

Reply:
Implemented in https://github.com/morpho-org/pre-liquidation/pull/37/commits/4b1dc0a6ac503942c35ac02ba1cccc52990fea4d.